sequencer/src/api/sql.rs

```diff
@@ -62,7 +65,7 @@ mod impl_testable_data_source {
     #[async_trait]
     impl TestableSequencerDataSource for DataSource {
         type Storage = TmpDb;
-        type Persistence = SqlStorage;
+        type Persistence = Arc<RwLock<SqlStorage>>;
```

Comment:
Curious why we needed this change? I thought the data source would always be `RwLock`ed at a higher level